astropy/table/column.py

```diff
@@ -271,11 +273,31 @@ def __reduce__(self):
 
         return reconstruct_func, reconstruct_func_args, state
 
+    def __getslice__(self, start, stop):
+        return self.__getitem__(slice(start, stop))
+
+    
     def __getitem__(self, item):
         if isinstance(item, INTEGER_TYPES):
             return self.data[item]  # Return as plain ndarray or ma.MaskedArray
+
+        elif isinstance(item, slice):
```

Comment:
So there is a little problem here... In #3930 (and #3929 which it superceded) we decided to remove `__getitem__` from the `BaseColumn` class because it slows single item access (and hence iterating over column values) by more than a factor of 10.  This is a fundamentally due to having to call into the Python layer instead of directly to the C version from ndarray.

I don't see any obvious way to get around this.  What do you think @mdboom ?  If not, we may need to just lose indices whenever you do indexing (slice, fancy indexing, whatever) on a `Column`.  We can still do this at the Table level, which is a more likely case where you really want the indices to follow along.